### PR TITLE
Corey/merge-friendly

### PR DESCRIPTION
**Pull request by dchw**
Small change to pull out what we can to make it a little more merge friendly.

### CODE REVIEW

File: session/configurabletimeout.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+				"timeout":        healthCfg.timeout,
+				"actualDuration": time.Since(timeoutStart),
+			}
+
+			if err != nil {
+				consecutiveFailures++
+
+				logFields["allowedFailures"] = healthCfg.allowedFailures
+				logFields["consecutiveFailures"] = consecutiveFailures
+				bklog.G(ctx).WithFields(logFields).Warn("healthcheck failed")
```
Comment (alexcb) [inline session/configurabletimeout.go:44]: is it safe to use `ctx` here, given it was previously `cancel()ed`?

**Review by alexcb — COMMENTED** [review]
thanks for moving this into a separate function; just one question regarding the use of the cancelled context.

File: session/configurabletimeout.go (hunk 1 above)
Comment (dchw) [reply session/configurabletimeout.go:44]: Seems to work on failures... I get the log messages out

**Review by vladaionescu — APPROVED** [review]

File: session/configurabletimeout.go (hunk 1 above)
Comment (alexcb) [reply session/configurabletimeout.go:44]: this still feels weird to me, someone might extend this code at a later date without realising the context is guaranteed to be cancelled.

Comment (alexcb) [reply session/configurabletimeout.go:44]: feel free to disregard this nitpick.

**Review by alexcb — APPROVED** [review]

**Pull request merged**